site/content/contribute/webapp/end-to-end-tests.md

```diff
@@ -81,42 +88,41 @@ Inside of the `integration` directory, there are additional directories that bre
-2. Add commands or shortcuts to `/e2e/cypress/support/commands.js` (e.g. `toAccountSettingsModal`) that make it easier to access a page, section, modal and so on by using it as `cy.toAccountSettingsModal()`.
+2. Add custom commands to `/e2e/cypress/support`. See Cypress documentation for more details about custom commands [[link](https://docs.cypress.io/api/cypress-api/custom-commands.html)].
+   - For ease of use, in-code documentation and discoverability, custom commands should have type definition added. See [declaration file](https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/support/api/user.d.ts) for reference on how to include.
 3. Organize `/e2e/cypress/integration` with a subfolder to group similar tests.
-4. Add a `Test Key` when describing a test.  In the spec file, it should be written as:
+4. Each test should have corresponding test case in Test Management for Jira (TM4J). Therefore, the `describe` block should correspond to folder name in TM4J (e.g. "Incoming webhook"), and `it` block should contain `TM4J number` as `Test Key` and test title (e.g. "MM-T623 Lock to this channel on webhook configuration works").  In the spec file, it should be written as:
 
     ```javascript
-    describe('Messaging', () => {
-        it('M15011 - Emojis show as jumbo in reply thread', () => {
+    describe('Incoming webhook', () => {
+        it('MM-T623 Lock to this channel on webhook configuration works', () => {
             // Test steps and assertion here
         }
     }
     ```
 
-    Where `"M15011"` is the key and `"Emojis show as jumbo in reply thread"` is the test description.
-
-    The naming convention used is the name of the tab on the release testing spreadsheet e.g. AS = Account Settings; M = Messaging, etc. and the numerical value is from Jira ticket identifier. So `"M15011"` means "Messaging" on release testing spreadsheet with `MM-15011` Jira ticket.
-
     For those writing E2E from help-wanted tickets with `Area/E2E Tests` label, the `Test Key` is available in the [GitHub issue itself](https://github.com/mattermost/mattermost-server/issues/10574).
 
-    `Test Key` is used to track test cases in our core staff Release Testing spreadsheet, which is linked in the header of the [Release Discussion channel](https://community.mattermost.com/core/channels/release-discussion) during release testing.
+    `Test Key` is used for mapping test cases per Release Testing specification. It will be used to measure coverage between manual and automated tests.
 
-    In cases where `Test Key` is not given, feel free to add a concise description of a test.
+    In case the `Test Key` is not available, feel free to prompt QA team who will either search from an existing TM4J entry or if it's a new one, it will be created for you.
```

Comment:
```suggestion
    In case the `Test Key` is not available, feel free to prompt the QA team who will either search from an existing TM4J entry or if it's a new one, it will be created for you.
```